File: day 8/tempCodeRunnerFile.cpp
#include <iostream>
using namespace std;

class BankAccount {
private:
    int accountNumber;
    double balance;

public:
    // Constructor to initialize account details
    BankAccount(int accNum, double bal) {
        accountNumber = accNum;
        balance = bal;
    }

    // Friend function declaration
    friend void displayAccountDetails(BankAccount);
};

// Friend function definition
void displayAccountDetails(BankAccount account) {
    cout << "Account Number: " << account.accountNumber << endl;
    cout << "Balance: $" << account.balance << endl;
}

int main() {
    // Create a BankAccount object with account number and balance
    BankAccount myAccount(123456, 2500.75);

    // Call the friend function to display account details
    displayAccountDetails(myAccount);

    return 0;
}

File: day 8/tempCodeRunnerFile.java

#include <iostream>

class Beta; // Forward declaration of class Beta

class Alpha {
private:
    int privateValue;

public:
    Alpha() : privateValue(42) {}

    // Declare Beta as a friend class
    friend class Beta;
};

class Beta {
public:
    void showPrivateValue(Alpha& a) {
        std::cout << "The private value of Alpha is: " << a.privateValue << std::endl;
    }
};

int main() {
    Alpha a;
    Beta b;
    b.showPrivateValue(a);

    return 0;
}